File: 0154-find-minimum-in-rotated-sorted-array-ii/0154-find-minimum-in-rotated-sorted-array-ii.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        s=0
        n=len(nums)
        e=n-1
        minn=float('inf')
        while(s<=e):
            mid=s+(e-s)//2
            if(nums[mid]<minn):
                minn=nums[mid]
            if(nums[mid]>nums[e]):
                s=mid+1
            elif(nums[mid]<nums[e]):
                e=mid-1
            else:
                e-=1
        return minn

File: 0154-find-minimum-in-rotated-sorted-array-ii/0154-find-minimum-in-rotated-sorted-array-ii.java
class Solution {
    public int findMin(int[] nums) {
        int s=0;
        int n=nums.length;
        int e=n-1;
        int minn=Integer.MAX_VALUE;
        while(s<=e)
        {
            int mid=s+(e-s)/2;
            if(nums[mid]<minn)
            {
                minn=nums[mid];
            }
            if(nums[mid]>nums[e])
            {
                s=mid+1;
            }
            else if(nums[mid]<nums[e])
            {
                e=mid-1;;
            }
            else
            {
                e--;
            }
        }
        return minn;
    }
}